core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java

```diff
@@ -33,23 +33,25 @@
   @Override
   public void accept(FluxSink<T> fluxSink) {
     FluxSinkRecorderDelegate<T> previousDelegate = this.delegate;
-    delegate = new DirectDelegate<>(fluxSink);
-    previousDelegate.accept(fluxSink);
+    synchronized (this) {
+      delegate = new DirectDelegate<>(fluxSink);
+      previousDelegate.accept(fluxSink);
+    }
   }
 
-  public FluxSink<T> getFluxSink() {
+  public synchronized FluxSink<T> getFluxSink() {
```

Comment:
have you meassured the perf impact of synchonizing all these methods?

Reply:
I'm on it. I've run the tests in 4.2.2